### PR TITLE
feat: Add block proB pod unless authpolicy present

### PR DESCRIPTION
**Pull request by wg102**
Closes https://github.com/StatCan/gatekeeper-policies/issues/15

Assumptions:
The pod inherits the `notebook-name` and classification labels.

The authorization policy is `{NOTEBOOK_NAME}-prob-block`

This blocks all the pods that are not unclassified. Not only the protected-b ones.
And it applies to all notebooks, it is not restricted to rstudio images since it is hard to detect which image it is.
The Authorization Policy will only be blocking certain URLs which are specific to rstudio.

cc @zachomedia

### CODE REVIEW

**Comment by wg102** [comment]
Updated the PR to be specifically about protected-b classification label since 

> If the classification label is not provided, then the workload is assumed to be Unclassified.

File: aaw/prob-auth/template.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  crd:
+    spec:
+      names:
+        kind: ProBAuth
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package probauth
+        violation[{"msg": msg}] {
+          input.review.object.metadata.labels["data.statcan.gc.ca/classification"] == "protected-b"
```
Comment (zachomedia) [inline aaw/prob-auth/template.yaml:15]: I think we can remove this from the check here and apply a selector to the constraint object. See https://github.com/StatCan/gatekeeper-policies/blob/feat-aaw-policies/aaw/metadata-restrictions/constraint.yaml#L29-L34 as an example of this.

**Review by zachomedia — CHANGES_REQUESTED** [review]
Thank you, at first glance this is looking good to me! I just have one change to request that isn't anything major!

**Review by blairdrummond — APPROVED** [review]
LGTM!

**Comment by blairdrummond** [comment]
@zachomedia Are you able to re-review?

**Comment by sylus** [comment]
I rebased this and pushed to master :D 

https://github.com/StatCan/gatekeeper-policies/commit/fb765123e5bbc2bd4e26811b6615c051d38fa9e9

**Pull request closed without merging**